File: app/src/main/java/com/example/bmi/QuizMain.java
package com.example.bmi;
import android.content.DialogInterface;
import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.github.mikephil.charting.charts.BarChart;
import com.github.mikephil.charting.data.BarData;
import com.github.mikephil.charting.data.BarDataSet;
import com.github.mikephil.charting.data.BarEntry;
import com.github.mikephil.charting.utils.ColorTemplate;

import java.util.ArrayList;
import java.util.Random;

public class QuizMain extends AppCompatActivity {

    Button answer1, answer2, answer3, answer4;
    TextView score, question;

    private QuizQuestions Questions = new QuizQuestions();

    private String Answer;
    private int Score = 0;
    private int QuestionLength = Questions.Questions.length;

    Random r;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.quiz_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        r = new Random();

        answer1 = (Button) findViewById(R.id.answer1);
        answer2 = (Button) findViewById(R.id.answer2);
        answer3 = (Button) findViewById(R.id.answer3);
        answer4 = (Button) findViewById(R.id.answer4);

        score = (TextView) findViewById(R.id.score);
        question = (TextView) findViewById(R.id.question);

        score.setText("Score: " + Score);

        updateQuestion(r.nextInt(QuestionLength));

        answer1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(answer1.getText().toString() == Answer){
                    Score++;
                    score.setText("Score: " + Score);
                    updateQuestion(r.nextInt(QuestionLength));
                }
                else {
                    gameOver();
                }
            }
        });

        answer2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(answer2.getText().toString() == Answer){
                    Score++;
                    score.setText("Score: " + Score);
                    updateQuestion(r.nextInt(QuestionLength));
                }
                else {
                    gameOver();
                }
            }
        });

        answer3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(answer3.getText().toString() == Answer){
                    Score++;
                    score.setText("Score: " + Score);
                    updateQuestion(r.nextInt(QuestionLength));
                }
                else {
                    gameOver();
                }
            }
        });

        answer4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(answer4.getText().toString() == Answer){
                    Score++;
                    score.setText("Score: " + Score);
                    updateQuestion(r.nextInt(QuestionLength));
                }
                else {
                    gameOver();
                }
            }
        });
    }

    private void updateQuestion(int num) {
        question.setText(Questions.getQuestion(num));
        answer1.setText(Questions.getAnswer1(num));
        answer2.setText(Questions.getAnswer2(num));
        answer3.setText(Questions.getAnswer3(num));
        answer4.setText(Questions.getAnswer4(num));

        Answer = Questions.getCorrectAnswer(num);
    }

    private void gameOver() {
        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(QuizMain.this);
        alertDialogBuilder
                .setMessage("GAME OVER!" + Score)
                .setCancelable(false)
                .setPositiveButton("NEW GAME",
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {
                                startActivity(new Intent(getApplicationContext(), QuizMain.class));
                            }
                        })
                .setNegativeButton("Exit",
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {
                                finish();
                            }
                        });
        AlertDialog alertDialog = alertDialogBuilder.create();
        alertDialog.show();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/example/bmi_kotlin/QuizMain.kt
package com.example.bmi_kotlin

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity


class QuizMain : AppCompatActivity() {

    var answer1: Button? = null; var answer2: Button? = null
    var answer3: Button? = null; var answer4: Button? = null
    var question:TextView? = null

    private val Questions = QuizQuestions()

    private var Answer: String? = null
    private var Score: Int = 0
    private val QuestionLength: Int = Questions.Questions.size

    var r: Int? = null
    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.quiz_main)
        setSupportActionBar(findViewById(R.id.toolbar))



        answer1 = findViewById(R.id.answer1)
        answer2 = findViewById(R.id.answer2)
        answer3 = findViewById(R.id.answer3)
        answer4 = findViewById(R.id.answer4)

        val score: TextView = findViewById(com.example.bmi_kotlin.R.id.score)
        question = findViewById(R.id.question)

        score.text = ("Score: $Score")

        updateQuestion((0..QuestionLength).random())
        answer1?.setOnClickListener {
            if(answer1?.text.toString() == Answer){
                Score++
                score.text = ("Score: " + Score)
                updateQuestion((0..QuestionLength).random())
            }
            else {
                gameOver()
            }
        }

        answer2?.setOnClickListener {
            if(answer2?.text.toString() == Answer){
                Score++
                score.text = ("Score: " + Score)
                updateQuestion((0..QuestionLength).random())
            }
            else {
                gameOver()
            }
        }

        answer3?.setOnClickListener {
            if(answer3?.text.toString() == Answer){
                Score++
                score.text = ("Score: " + Score)
                updateQuestion((0..QuestionLength).random())
            }
            else {
                gameOver()
            }
        }

        answer4?.setOnClickListener {
            if(answer4?.text.toString() == Answer){
                Score++
                score.text = ("Score: " + Score)
                updateQuestion((0..QuestionLength).random())
            }
            else {
                gameOver()
            }
        }
    }

    private fun updateQuestion(num: Int) {
        question?.text = Questions.getQuestion(num)
        answer1?.text = Questions.getAnswer1(num)
        answer2?.text = Questions.getAnswer2(num)
        answer3?.text = Questions.getAnswer3(num)
        answer4?.text = Questions.getAnswer4(num)

        Answer = Questions.getCorrectAnswer(num)
    }

    private fun gameOver() {
        val alertDialogBuilder = AlertDialog.Builder(this@QuizMain)
        alertDialogBuilder
            .setMessage("GAME OVER! Your score is: $Score")
            .setCancelable(false)
            .setPositiveButton(
                "New game"
            ) { dialog, which -> startActivity(Intent(applicationContext, QuizMain::class.java)) }
            .setNegativeButton(
                "Exit"
            ) { dialog, which -> finish() }
        val alertDialog = alertDialogBuilder.create()
        alertDialog.show()
    }
}